src/slha_format.hpp

```diff
@@ -38,6 +40,25 @@ extern const char * const single_element_formatter;
 /// SLHA line formatter for the SPINFO block entries
 extern const char * const spinfo_formatter;
 
+template <typename Container>
+boost::format format_decay(double br, const Container& pids, const std::string& name)
+{
+   const int nda = pids.size();
+
+   boost::format formatted_ids("");
+   for (int i = nda - 1; i >= 0; --i) {
+      formatted_ids = boost::format(" %9d%s") % pids[i] % formatted_ids;
+   }
+
+   return boost::format("   %16.8E  %2d  %s  # %s\n") % br % nda % formatted_ids % name;
+}
+
+
+inline boost::format format_total_width(int pdg, double width, const std::string& name)
+{
+   return boost::format("%9d   %16.8E   # %s\n") % pdg % width % name;
+}
+
```

Comment:
@Expander, Dylan added extra formatting functions. You recently made a change to the way how we do it? Should we also change this?

Reply:
Yes, please.

My recent change had the following reasons:

1. It improves compile time, because `boost/format.hpp` is no longer included. Furthermore the format strings are now defined in a `.cpp` file (and not in a header), which in addition improves compile time.
2. I have replaced `boost::format()` by `snprintf()`, which is significantly faster, see https://www.boost.org/doc/libs/1_67_0/libs/format/doc/format.html#alternatives

So, from my point it has more benefits than disadvantages. :)

Reply:
I've fixed the `format_total_width`. I'm not sure what to do with `format_decay` though. It has a `boost::format` loop. I wouldn't mind if you'd take a look at it :)

Reply:
I'd propose to rewrite the `format_decay` function as follows:
~~~
template <typename Container>
std::string format_decay(double br, const Container& pids, const std::string& name)
{
   const int nda = pids.size();
   char buf[SLHA_MAX_LINE_LENGTH] = { 0 };
   int written = 0;

   written += std::snprintf(buf + written, SLHA_MAX_LINE_LENGTH - written,
                            "   %16.8E  %2d  ", br, nda);

   for (int i = 0; i < nda; i++) {
      written += std::snprintf(buf + written, SLHA_MAX_LINE_LENGTH - written,
                               " %9d", pids[i]);
   }

   written += std::snprintf(buf + written, SLHA_MAX_LINE_LENGTH - written,
                            "  # %s\n", name.c_str());

   return std::string(buf);
}
~~~

Reply:
Perfect. Thanks!